xmake/rules/c++/modules/modules_support/gcc/builder.lua

```diff
@@ -387,9 +387,10 @@ function make_headerunit_buildjobs(target, job_name, batchjobs, headerunit, bmif
                     if option.get("diagnosis") then
                         print("mapper file:\n%s", io.readfile(headerunit_mapper))
                     end
+                    local headerfile = headerunit.unique and headerunit.name or headerunit.path
                     _compile(target,
                         _make_headerunitflags(target, headerunit, headerunit_mapper, opt),
-                        path.translate(path.filename(headerunit.name)), bmifile)
```

Comment:
we can not use path.filename here for gcc. if we use packages in modules. like spdlog package. `#import <spdlog/spdlog.h>;`

```console
/usr/bin/gcc -c -m64 -fvisibility=hidden -fvisibility-inlines-hidden -O3 -std=c++20 -isystem /home/ruki/.xmake/packages/s/spdlog/v1.15.2/51f8d8b9d14d4d17af0b7c28620bf3c7/include -D_GLIBCXX_USE_CXX11_ABI=0 -DNDEBUG -fmodule-mapper=/tmp/.xmake999/250414/_F91056BC7F6C48108C5C73EE27ED1330 -fmodule-header=system -fmodule-only -xc++-header -x c++-header -o build/.gens/mod/linux/x86_64/release/rules/bmi/cache/headerunits/spdlog.h.gcm spdlog.h
cc1plus: fatal error: spdlog.h: No such file or directory
compilation terminated.

```